e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { Locator, Page, expect, test as baseTest } from '@playwright/test';
+import { Portal } from 'e2e/support/fixtures';
+
+const TEST_SETUPLINK_REDIRECT_URL = 'http://localhost:3366';
+const TEST_SETUPLINK_DEFAULT_REDIRECT_URL = 'http://localhost:3366/login/saml';
+const TEST_SETUPLINK_MOCK_METADATA_URL = 'https://mocksaml.com/api/namespace/setuplink-test/saml/metadata';
+const TEST_SETUPLINK_URL_LABEL_SELECTOR =
+  'Share this link with your customers to allow them to set up the integrationClose';
+const TEST_SETUPLINK_ADMIN_URL = '/admin/sso-connection/setup-link';
+const TEST_SETUPLINK_ADMIN_CONNECTION = '/admin/sso-connection';
+
+type MyFixtures = {
+  portal: Portal;
+};
+
+export const test = baseTest.extend<MyFixtures>({
+  portal: async ({ page }, use) => {
+    const portal = new Portal(page);
+    await portal.doCredentialsLogin();
+    await use(portal);
+  },
+});
+
+async function retryForCondition(condition: Promise<any>, retries = 10, retryInterval = 500) {
+  while (retries) {
+    if (await condition) {
+      break;
+    } else {
+      setTimeout(() => {
+        retries -= 1;
+      }, retryInterval);
+    }
+  }
+}
+
+async function getRowByTenantProduct(
+  page: Page,
+  tenant: string,
+  product: string,
+  tenantProductIndexes: Array<number>
+): Promise<Locator | string> {
+  const rows = await page.getByRole('row').all();
+  const setupLinks = await Promise.all(rows.map(async (el: Locator) => el));
+  let row: Locator | string = '';
+
+  for (const link of setupLinks) {
+    const el = await link.innerText();
+    const cellContents = el.split('\t');
+    if (
+      cellContents.length > 1 &&
+      cellContents[tenantProductIndexes[0]] === tenant &&
+      cellContents[tenantProductIndexes[1]] === product
+    ) {
+      row = link;
+    }
+  }
+
+  return row;
+}
+
+test.describe('Admin Portal Enterprise SSO SetupLink using generic SAML 2.0', () => {
+  test('should be able to create setup link and sso connection using generic SAML 2.0', async ({ page }) => {
+    const tenant = 'acme-setuplink-test.com';
+    const product = 'acme-setuplink-test';
+
+    // Go to admin/sso-connection/setup-link page and create setup link
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+    await page.getByRole('button', { name: 'New Setup Link' }).click();
+    await page.locator('input[type="text"]').nth(0).fill('acme-test');
```

Comment:
Instead of using `nth` here, it's [better](https://playwright.dev/docs/locators#get-by-nth-item:~:text=However%2C%20use%20this%20method%20with%20caution.%20Often%20times%2C%20the%20page%20might%20change%2C%20and%20the%20locator%20will%20point%20to%20a%20completely%20different%20element%20from%20the%20one%20you%20expected.%20Instead%2C%20try%20to%20come%20up%20with%20a%20unique%20locator%20that%20will%20pass%20the%20strictness%20criteria.) to use a unique locator like `getByLabel` or `getByPlaceholder`
> However, use this method with caution. Often times, the page might change, and the locator will point to a completely different element from the one you expected. Instead, try to come up with a unique locator that will pass the [strictness criteria](https://playwright.dev/docs/locators#strictness).